compileProtobuffers.ts

```diff
@@ -0,0 +1,65 @@
+import chalk from "chalk";
+import fs from "fs-extra";
+import { exec, execSync } from "node:child_process";
+
+const PROTOBUFF_DIR = "./Ecdar-ProtoBuf/";
+const OUT_DIR = "./src/lib/protobuf/";
+
+export const compileProtobuffers = {
+	name: "Compiling protobuffers",
+
+	buildStart: async () => {
+		await isClosed();
+		await fs.ensureDir(OUT_DIR);
+		const res = await Promise.all(
+			(await fs.readdir(PROTOBUFF_DIR))
+				.filter((file) => file.match(/.*\.proto/g))
+				.map((file) =>
+					runcmd(`
+					  yarn protoc \\
+						  --ts_out ${OUT_DIR} \\
+						  --proto_path ${PROTOBUFF_DIR} \\
+						  ${PROTOBUFF_DIR}${file}
+					`),
+				),
+		);
+
+		if (res.length === 0) {
+			const err = `Failed to compile protobuffers, ${PROTOBUFF_DIR} is empty`;
+			console.error(`${chalk.red("❌")}${err}`);
+			throw new Error(err);
+		}
+		console.log(`${chalk.green("✔")} Compiled Protobuffers: Done`);
+	},
+};
+
+function runcmd(cmd: string): Promise<void> {
+	return new Promise((res, rej) => {
+		exec(cmd, (err, stdout, stderr) => {
+			if (err !== null) {
+				const reducedCmd = cmd
+					.replace(/\\.*\n/g, "")
+					.replace(/\s+/g, " ");
+				console.error(`Protobuff command failed\ncmd:\n${reducedCmd} `);
+				console.error(`stdout:\n${stdout}`);
+				console.error(`stderr:\n${stderr}`);
+				rej();
+			}
+			res();
+		});
+	});
+}
+
+function isClosed(): Promise<void> {
+	return new Promise((res) => {
+		for (;;) {
+			try {
+				execSync("yarn protoc --version");
+				res();
+				break;
+			} catch {
+				[]; // NO EMPTY BLOCK STATEMENTS
```

Comment:
It is not OK to circumvent the linter like this.

What happens if the protoc install fails or if you make a mistake in the code that means this check will never be true? A GitHub runner will run idle for many hours until GitHub kills it. That is very expensive.  A user running it on their local system would be waiting in vain while the script gives them no feedback that something is wrong. 

In this block you should be checking to see how long ago the first attempt was made. If it is more than a few minutes ago, I would throw an error message that nicely explains what happened. The wait time should be easily configurable in a `const`. That way if a user really does have a very slow PC, the error message will explain exactly what happened, and they will easily be able to find this file and extend the wait time.